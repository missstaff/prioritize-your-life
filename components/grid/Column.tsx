import { ScaledSheet } from "react-native-size-matters";
import AppThemedView from "../app_components/AppThemedView";

interface ColumnProps {
  children: React.ReactNode;
  inlineStyles?: object;
}

const Column = ({ children, inlineStyles, ...props }: ColumnProps) => {
  return (
    <AppThemedView style={[styles.column, inlineStyles]} {...props}>
      {children}
    </AppThemedView>
  );
};

const styles = ScaledSheet.create({
  column: {
    alignItems: "flex-start",
    display: "flex",
    flexDirection: "column",
    flexGrow: 1,
    justifyContent: "center",
    width: "33%",
  },
});

export default Column;
