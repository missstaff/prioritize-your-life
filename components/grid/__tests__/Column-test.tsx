import React from "react";
import { Text } from "react-native";
import { render } from "@testing-library/react-native";
import Column from "../Column";

describe("Column Component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly with no children", () => {
    const { toJSON } = render(<Column children={undefined} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly with children", () => {
    const { getByText } = render(
      <Column>
        <Text>Test Child</Text>
      </Column>
    );
    expect(getByText("Test Child")).toBeTruthy();
  });
});
