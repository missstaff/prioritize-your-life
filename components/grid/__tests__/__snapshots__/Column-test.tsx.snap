// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column Component renders correctly with no children 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#151718",
      },
      [
        {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "column",
          "flexGrow": 1,
          "justifyContent": "center",
          "width": "33%",
        },
        undefined,
      ],
    ]
  }
/>
`;
